llvm/lib/Target/AMDGPU/AMDGPUAttributor.cpp

```diff
@@ -728,12 +735,43 @@ struct AAAMDSizeRangeAttribute
     };
 
     bool AllCallSitesKnown = true;
-    if (!A.checkForAllCallSites(CheckCallSite, *this, true, AllCallSitesKnown))
+    if (!A.checkForAllCallSites(CheckCallSite, *this,
+                                /*RequireAllCallSites=*/true,
+                                AllCallSitesKnown))
       return indicatePessimisticFixpoint();
```

Comment:
This will make the AA invalid as well because if `updateImpl` is called, the known range is full set for sure, and `known -> assumed` will make the assumed range a full set, thus in an invalid state.